seahorn/trusty/user/base/lib/CMakeLists.txt

```diff
@@ -0,0 +1,12 @@
+set(EXTERNAL ${TRUSTY_ROOT}/trusty/user/base/lib/libc-trusty)
+add_library(
+  sea_libc_trusty
+  ${EXTERNAL}/ipc.c
+)
+
+if (HANDLE_TYPE_IS_PTR)
+  add_subdirectory(libc-trusty-ptr)
+else ()
+  add_subdirectory(libc-trusty)
+endif ()
+add_compile_definitions(CAN_RETURN_INVALID_IPC_HANDLE=0)
```

Comment:
are there multiple conflicting compiler definitions? Are you assuming they are resolved somehow by CMake or by the compiler?

Reply:
In the original spec (impl), some jobs are written assuming a valid `handle_t` is always returned by the environment. Other jobs account for failure. I have added this `compile_definition` to control what the environment will return per job.

Reply:
I noticed that this definition is added multiple times in different CMakeLists.txt files. I am asking how the exclusivity of which definition is active is maintained.

Reply:
The short answer is that exclusivity is maintained by user discipline. Each directory of jobs is independent of other jobs hence there should not be a conflict. 

--
The problem is that this can cause silent unexpected behaviour if user breaks discipline or jobs are rearchitected to be dependent on each other. One solution is to move to the `target` based compilation model instead of the `directory` one used now. In doing so we can ensure that dependencies are explicit and there are known (few) points where a `token` can redefined. Right now, each directory can redefine a `token` which can become hard to reason about.

Changing the compilation model seems out of scope in this PR.